tcms/xmlrpc/api/telemetry.py

```diff
@@ -0,0 +1,64 @@
+from modernrpc.core import rpc_method
+
+from tcms.management.models import Product
+from tcms.testcases.models import TestCase
+from tcms.testplans.models import TestPlan
+
+
+@rpc_method(name='Telemetry.get_data')
+def get_data(request):
+
+    test_plan_id = request.get('test_plan_id')
+    if not test_plan_id:
+        raise Exception('Required test_plan_id')
+
+    test_plan = TestPlan.objects.get(plan_id=test_plan_id)
+
+    test_cases = TestCase.objects.filter(plan=test_plan)
+
+    count = {
+        'manual': 0,
+        'automated': 0
+    }
+    for test_case in test_cases:
+        if test_case.is_automated:
+            count['manual'] += 1
+        else:
+            count['automated'] += 1
+    count['all'] = count['automated'] + count['manual']
+
+    priorities = {}
+    for test_case in test_cases:
+        if priorities.get(test_case.priority.value):
+            priorities[test_case.priority.value] += 1
+        else:
+            priorities[test_case.priority.value] = 1
+
+    priorities_arr = []
+    for key, value in priorities.items():
+        priorities_arr.append([key, value])
+
+    categories = {}
+    for test_case in test_cases:
+        if categories.get(test_case.category.name):
+            categories[test_case.category.name] += 1
+        else:
+            categories[test_case.category.name] = 1
+
+    categories_arr = []
+    for key, value in categories.items():
+        categories_arr.append([key, value])
+
+    statuses = {}
+    for test_case in test_cases:
+        if statuses.get(test_case.case_status.name):
+            statuses[test_case.case_status.name] += 1
+        else:
+            statuses[test_case.case_status.name] = 1
```

Comment:
Same thing for aggregation/counting for all of these fields. You can then take the values from the QuerySet and return them in the dict below.